cmd/readiness/readiness_test.go

```diff
@@ -54,6 +55,34 @@ func TestNotReadyHealthFileHasNoProcesses(t *testing.T) {
 	assert.False(t, isPodReady(testConfig("testdata/health-status-no-processes.json")))
 }
 
+// TestNotReadyHealthFileAbsent verifies that the readiness logic panics is the health file is absent.
+// Note, that historically the absence of the file was interpreted as "ready" when older version of agents
+// could be used. But now all supported Agent versions always produce this file.
+func TestNotReadyHealthFileAbsent(t *testing.T) {
+	assert.Panics(t, func() { isPodReady(testConfig("non-existing.json")) })
```

Comment:
rather than asserting panics, maybe we could even change the signature of `isPodReady` to return `(bool, error)` I believe this would be more idiomatic and also allow to differentiate between errors and `false` because we have just not reached ready state.

Reply:
yes, this makes sense, agree 👍